### DISCUSSION TITLE
c-ares governance

### DISCUSSION THREAD

**Discussion by yanavlasov** [c-ares/c-ares#919] [Q&A]
Hello c-ares maintainers,
are there any plans to tighten governance of the c-ares library? Specifically the commit policy that requires maintainer approval for all PRs. A large number of commits in c-ares repo are pushed directly, without PR or review. This causes violations of dependency policies for a number of projects that depend on c-ares.

**Comment by bradh352** [comment]
At this point in time, there is a single developer working on c-ares, me.  So not exactly sure how we could implement any sort of review process.

**Reply by bradh352** [reply]
We could however ensure all commits come from PRs, but not sure how useful that really is.  There are some things that PRs won't work for, like fixing code smells in SonarCloud.

Every commit does go through all our test suites on multiple platforms with dynamic analysis, and also runs through a few different static analyzers.  Our test suite has > 90% code coverage.  We also are part of google's OSS Fuzz.

Having only a single maintainer is really an issue.  I can pull Daniel (@bagder) in on major things, but he's really busy with curl and hasn't dedicated much time to c-ares in a long time.

**Reply by yanavlasov** [reply]
Thanks for the explanation. I'm looking on my side for ways to help here.

**Reply by bradh352** [reply]
can you state what the governance requirements are that are blockers and maybe what other small OSS projects do to try to comply?
